robot_driver/include/robot_driver/estimators/state_estimator.h

```diff
@@ -0,0 +1,95 @@
+#ifndef STATE_ESTIMATOR_H
+#define STATE_ESTIMATOR_H
+
+#include <quad_msgs/ContactMode.h>
+#include <quad_msgs/RobotState.h>
+#include <quad_utils/quad_kd.h>
+#include <quad_utils/ros_utils.h>
+#include <ros/ros.h>
+#include <sensor_msgs/Imu.h>
+#include <sensor_msgs/JointState.h>
+#include <std_msgs/String.h>
+
+#include "robot_driver/hardware_interfaces/hardware_interface.h"
+#include "robot_driver/hardware_interfaces/spirit_interface.h"
+
+//! Implements an abstract class for state estimator.
+//! This class provides an interface for different types of estimators
+/*!
+   StateEstimator provides an abstract state estimator class
+*/
+class StateEstimator {
+ public:
+  /**
+   * @brief Constructor for StateEstimator
+   * @return Constructed object of type StateEstimator
+   */
+  StateEstimator();
+
+  /**
+   * @brief virtual function for initialize filters, should be defined in
+   * derived class
+   * @param[in] nh_ sensor_msgs::Imu::ConstPtr imu sensor message
+   */
+  virtual void init(ros::NodeHandle& nh) = 0;
+
+  /**
+   * @brief virtual update function for update robot state, should be defined in
+   * derived class
+   * @param[out] last_robot_state_msg quad_msgs::RobotState robot state
+   */
+  virtual bool updateOnce(quad_msgs::RobotState& last_robot_state_msg) = 0;
+
+  /**
+   * @brief read IMU data
+   * @param[in] last_imu_msg sensor_msgs::Imu::ConstPtr imu sensor message
+   * @param[out] fk Eigen::Vector3d linear acceleration (3 * 1)
+   * @param[out] wk Eigen::Vector3d angular acceleration (3 * 1)
+   * @param[out] qk Eigen::Quaterniond orientation in quaternion
```

Comment:
You don't need to list the types in the comments, Doxygen will handle that (apply this to other function comments in this PR).